### ISSUE TITLE
[v2][Question] Filter pipe needed, mine is just hacky, will it be supported?

### ISSUE THREAD

**Issue by Luchillo** [angular/angular#6707]
I made my own, which supports array of arrays recursive, array of strings and array of objects with 1 level nesting, haven't implemented expressions, just fuzzy string search.

Will there be a filter pipe ever? i think mine is just a little hacky.

**Comment by laco0416** [comment]
:+1: 

If it'll be supported, I want use expressions for filtering.
```
<div *ngFor="#item of items | filter: someExpression(item)">...</div>

...

someExpression(item) boolean {
  ...
}
```

**Comment by woppa684** [comment]
You mean something like this ... ? 
http://plnkr.co/edit/XCA88dyeSnRF0JgA7cSU?p=preview

**Comment by Luchillo** [comment]
Yes, just want something like the filter pipe in Angular 1 where you could use expressions, regular expressions and even functions.

**Comment by pjhartin** [comment]
I should hae searched here earlier. I spent approx. 4hrs today trying to implement this. New to JS and couldn't solve it! Does the plnkr.co link explain how to implement this?

**Comment by Luchillo** [comment]
Yep, i get it, it's bad for the framework to provide such a pipe, since the Ng1 filter pipe awarded the framework a slow reputation.

I guess it is better to the developer to have specialized and **fast** filter pipes for their specific use case, at least the pipe api is well documented.
